syncer/syncer.go

```diff
@@ -2742,8 +2742,43 @@ func (s *Syncer) createDBs(ctx context.Context) error {
 		if err2 != nil {
 			s.tctx.L().Warn("cannot get sql_mode from upstream database", log.ShortError(err2))
 		} else {
-			s.cfg.To.Session["sql_mode"] = sqlMode
-		}
+			// When upstream's datatime is 2020-00-00, 2020-00-01, 2020-06-00
+			// and so on, downstream will be 2019-11-30, 2019-12-01, 2020-05-31,
+			// as if set the 'NO_ZERO_IN_DATE', 'NO_ZERO_DATE'.
+			// This is because the implementation of go-mysql, that you can see
+			// https://github.com/go-mysql-org/go-mysql/blob/master/replication/row_event.go#L1063-L1087
+			sqlModes := strings.Split(sqlMode, ",")
+			needDisable := []string{
+				"NO_ZERO_IN_DATE",
+				"NO_ZERO_DATE",
+				"ERROR_FOR_DIVISION_BY_ZERO",
+				"NO_AUTO_CREATE_USER",
+				"STRICT_TRANS_TABLES",
+				"STRICT_ALL_TABLES",
+			}
+			for _, disable := range needDisable {
+				for i, sqlMode := range sqlModes {
+					if disable == sqlMode {
+						sqlModes = append(sqlModes[:i], sqlModes[i+1:]...)
```

Comment:
maybe parser has an elegant way to check and modify SQL modes

Reply:
I have thought about this problem, but I did not think of a better way. How about replacing and adding string directly?

```go
for _, disable := range needDisable {
	strings.RelpaceAll(l.cfg.LoaderConfig.SQLMode, disable, "")
}
for _, enable := range needEnable {
	l.cfg.LoaderConfig.SQLMode += "," + enable
}
```

Reply:
maybe https://github.com/pingcap/dm/blob/8089f1a3ba31f4a73e64e3346571e4ac46c5e6d8/pkg/utils/db.go#L429

Reply:
I have thought about this problem but did not think of a better way. How about replace and add string directly like this:
```go
for _, disable := range needDisable {
	l.cfg.LoaderConfig.SQLMode = strings.ReplaceAll(l.cfg.LoaderConfig.SQLMode, disable, "")
}
for _, enable := range needEnable {
	l.cfg.LoaderConfig.SQLMode += "," + enable
}
```

Reply:
I just find Str2SQLMode https://github.com/pingcap/parser/blob/master/mysql/const.go#L469. Should I add SQLMode2Str func?

Reply:
OK, I think parser will reject adding this function, we can add SQLMode2Str in DM

Reply:
So, should I open a new pr to add this function. I think I also should add a unit test for it?